### PR TITLE
Activity utils

### PR DESCRIPTION
**Pull request by Bjvanminnen**
I'm going to be making some changes to the course progress UI related to locked stages. As part of that, I'd like to refactor how some of our redux stuff here looks (both to make it look similar to our redux elsewhere, and to make clearer lines between our views, actions, and reducers).

As an initial step towards that, I found a couple of utility type methods that (a) don't need to be on the global namespace and (b) seem like they belong in their own module.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
@joshlory @mehalshah I think the two of you likely have the most knowledge/vested interest in this stuff right now. Let me know what you think of this.

File: apps/src/code-studio/activityUtils.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * Returns the "best" of the two results, as defined in apps/src/constants.js.
+ * Note that there are negative results that count as an attempt, so we can't
+ * just take the maximum.
+ * @param {Number} a
+ * @param {Number} b
+ * @return {Number} The better result.
+ */
+export const mergeActivityResult = (a, b) => {
```
Comment (mehalshah) [inline apps/src/code-studio/activityUtils.js:44]: can you just have a and b default to 0 here?

File: apps/src/code-studio/activityUtils.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  a = a || 0;
+  b = b || 0;
+  if (a === 0) {
```
Comment (mehalshah) [inline apps/src/code-studio/activityUtils.js:47]: You can probably make this terser

if (a === 0 || b === 0) {
return a || b
} else {
return Math.max(a, b)
}

**Comment by mehalshah** [comment]
Couple comments, rest LGTM

File: apps/src/code-studio/activityUtils.js (hunk 2 above)
Comment (Bjvanminnen) [inline apps/src/code-studio/activityUtils.js:47]: This is code that I moved from elsewhere (and was not the author of). As such, I'm somewhat wary of changing it and think I prefer to leave it as is.

**Pull request merged**